vhdbuilder/packer/test/run-test.sh

```diff
@@ -6,6 +6,11 @@ TEST_RESOURCE_PREFIX="vhd-test"
 TEST_VM_ADMIN_USERNAME="azureuser"
 TEST_VM_ADMIN_PASSWORD="TestVM@$(date +%s)"
 
+if [ "$OS_TYPE" == "Linux" ] && ([ "$OS_SKU" == "CBLMariner" ] || [ "$OS_VERSION" == "16.04" ] || [ "$MODE" == "gen2Mode" ]); then
```

Comment:
- Windows has no definition of OS_SKU.
- move the check before creating resource group